strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/AuthenticatorsRegistry.java

```diff
@@ -0,0 +1,139 @@
+package org.carlspring.strongbox.authentication.registry;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+import org.carlspring.strongbox.authentication.registry.support.AuthenticatorsRegistrySerializer;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@JsonSerialize(using = AuthenticatorsRegistrySerializer.class)
+public class AuthenticatorsRegistry
+        implements Supplier<List<Authenticator>>
+{
+
+    final transient ReentrantLock lock = new ReentrantLock();
+
+    private final List<Authenticator> authenticators;
```

Comment:
`authenticators` migth be SpringBeans but might be not. It's acceptable that some of them are SpringBeans, some are not.

Please take a look here https://github.com/fuss86/strongbox/blob/SB-625/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java#L84 

Generally, if the authenticator listed in the configuration file (https://github.com/fuss86/strongbox/blob/SB-625/strongbox-user-management/src/main/resources/etc/conf/strongbox-authentication-providers.xml) contains `component-scan-base-packages` attribute, it is considered to be a SpringBean. Otherwise custom instance is created (https://github.com/fuss86/strongbox/blob/SB-625/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java#L91) and empty constructor is required.